File: src/test/ruby/soap_test.rb
require "test/unit"
#require 'minitest/reporters'
#MiniTest::Reporters.use!

require 'java'
p $CLASSPATH
p $LOAD_PATH

java_import 'kravmaga.SoapServiceSimulator'
java_import 'kravmaga.generated.bookingservice.Conference'
java_import 'kravmaga.generated.bookingservice.ArrayOfParticipant'

class SoapTest < Test::Unit::TestCase

  # Called before every test method runs. Can be used
  # to set up fixture information.
  def setup
    # Do nothing
  end

  # Called after every test method runs. Can be used to tear
  # down fixture information.

  def teardown
    # Do nothing
  end

  def test_create
    client = SoapServiceSimulator.new
    conference = Conference.new
    conference.setStartTimeUTC('2012-10-10 12:00:00Z')
    conference.setEndTimeUTC('2012-10-10 13:00:00Z')
    conference.setParticipants(ArrayOfParticipant.new)
    p client.save_conference(conference)
  end
end

File: src/test/java/kravmaga/SoapTest.java
package kravmaga;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

import kravmaga.generated.bookingservice.ArrayOfParticipant;
import kravmaga.generated.bookingservice.Conference;
import kravmaga.generated.bookingservice.Participant;

public class SoapTest {

    @Test
    public void testCreate() {
        SoapServiceSimulator simulator = new SoapServiceSimulator();
        Conference conference = new Conference();
        conference.setStartTimeUTC("2012-10-10 12:00:00Z");
        conference.setEndTimeUTC("2012-10-10 13:00:00Z");
        conference.setParticipants(new ArrayOfParticipant() {
            @Override
            public List<Participant> getParticipant() {
                return new ArrayList<Participant>();
            }
        });
        simulator.saveConference(conference);
    }
}
